### ISSUE TITLE
bug(material/list, material/autocomplete): selected state is visually communicated only with color (WCAG 1.4.1)

### ISSUE THREAD

**Issue by zarend** [angular/components#25961]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

select visually communicate selected state with color alone. It should be visually communicated with more than just color. this also happens with autocomplete

https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-without-color.html

### Reproduction

Steps to reproduce:
1. open https://material2-dev.web.app/select
2. Navigate to "ngModel" demo
3. Click "Drink"
4. (popup appears with drink options)
5. Click "Coke"
6. (popup closes)
7. click "Coke"
8. (popup reopens)

### Expected Behavior

selected state of "Oranges" is visually communicated with more than color alone

### Actual Behavior

Color of "Coke" row changes to indicate selection. Selection is visually communicated with color alone. This is difficult to discern especially for end users who are colorblind or with low vision. (WCAG 1.4.1)
<img width="569" alt="image" src="https://user-images.githubusercontent.com/7720245/201432417-b00971cc-3802-464d-a7b8-2407c34ea0b1.png">

### Environment

N/A
